presto-hive/src/main/java/com/facebook/presto/hive/HiveDirectoryContext.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive;
+
+import static java.util.Objects.requireNonNull;
+
+public class HiveDirectoryContext
+{
+    private final NestedDirectoryPolicy nestedDirectoryPolicy;
+    private final boolean cacheable;
```

Comment:
nit: I find cacheable too generic, may be specify its for listDirectoryCache?

Reply:
I'm a bit hesitated because it might be somewhat tautologic given the class name is already  `HiveDirectoryContext`. What do you think?

Reply:
Not feeling strongly, but the name DirectoryContext, doesn't tell me that its for ListingDirectories.
Actually in future, I would like to see a generic context being used for all kinds of caching. With this change we would have got two contexts representing different caching information.

Reply:
I agree that consolidating to one context used for all kinds of caching is ideal. But there is also nuances between coordinator cache and worker cache, for whom we probably would still want some differentiation.

Currently the `cacheable` flag in `HiveFileContext` is determined by the node selection strategy, which is valid because file cache is on the workers, so it makes little sense to cache when there is no affinity. It does not feel natural to apply the same logic for listFiles, because it is coordinator-only caching, which means it is always hard affinity :P.

Let me keep it as is to be consistent with `HiveFileContext`. We could revisit when we try to consolidate cache context :)